src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -358,23 +370,139 @@ def setClean(self, wflow):
+        targetStatusList = RequestStatus.REQUEST_STATE_TRANSITION.get(wflow['RequestStatus'], [])
+        targetStatusReg = re.compile(r'.*archived')
+        for status in targetStatusList:
+            if targetStatusReg.match(status):
+                wflow['TargetStatus'] = status
+        self.logger.debug("TargetStatus: %s", wflow['TargetStatus'])
+        return wflow
+
+    def _cleanLogDB(self, wflow):
+        """
+        A function to be used for cleaning all the records related to a workflow in logDB.
+        :param wflow:       A MSRuleCleaner workflow representation
+        :return:            Bool
+        """
+        cleanStatus = False
+        try:
+            self.logger.info("Deleting %s from LogDB WMStats...", wflow['RequestName'])
+            res = self.logDB.delete(wflow['RequestName'], agent=False)
+            if res == 'delete-error':
+                self.logger.error("Failed to delete logDB docs for wflow: %s", wflow['RequestName'])
+            cleanStatus = True
+        except Exception as ex:
+            msg = "General Exception while cleaning LogDB records for wflow: %s : %s"
+            self.logger.exception(msg, wflow['RequestName'], str(ex))
+        return cleanStatus
+
+    def _checkArchDelayExpired(self, wflow):
+        """
+        A function to check Archival Expiration Delay based on the inromation
+        returned by WMStatsServer regarding the time of the last request status transition
+        :param wflow:      MSRuleCleaner workflow representation
+        :return:           Bool
+        """
+        archDelayExpired = False
+        currentTime = int(time.time())
+        threshold = self.msConfig['archiveDelayHours'] * 3600
+        outputMask = ['RequestTransition']
+        inputConditon = {'RequestName': wflow['RequestName']}
+        try:
+            wmsReqInfo = self.wmstatsSvc.getFilteredActiveData(inputConditon, outputMask)
+            # self.logger.debug("wmsReqInfo: \n%s", pformat(wmsReqInfo))
+            if wmsReqInfo and (currentTime - wmsReqInfo[0]["RequestTransition"][-1]["UpdateTime"]) > threshold:
+                archDelayExpired = True
+        except Exception as ex:
+            msg = "General Exception while checking the time of the last request status transition for wflow: %s : %s"
+            self.logger.exception(msg, wflow['RequestName'], str(ex))
+        return archDelayExpired
+
+    def setArchDelayExpired(self, wflow):
+        """
+        A function to set the 'IsArchDelayExpired' flag
+        """
+        wflow['IsArchDelayExpired'] = self._checkArchDelayExpired(wflow)
+        return wflow
+
+    def _archive(self, wflow):
+        """
+        A function to make the actual request status transition during the
+        archival step.
+        :param wflow:       A MSRuleCleaner workflow representation
+        :return:            Bool
+        """
+        archiveStatus = False
```

Comment:
I somehow have the feeling that we are overloading the usage of "private" methods in this module, basically every method has its own private method to execute the action. IMO, I'd move this logic under the `archive` method itself.

Reply:
Done